internal/ssa2ast/type.go

```diff
@@ -53,14 +53,21 @@ func (tc *typeConverter) Convert(t types.Type) (ast.Expr, error) {
 		return chanExpr, nil
 	case *types.Interface:
 		methods := &ast.FieldList{}
+		hasComparable := false
 		for i := 0; i < typ.NumEmbeddeds(); i++ {
 			embeddedType := typ.EmbeddedType(i)
+			if namedType, ok := embeddedType.(*types.Named); ok && namedType.String() == "comparable" {
+				hasComparable = true
+			}
 			embeddedExpr, err := tc.Convert(embeddedType)
 			if err != nil {
 				return nil, err
 			}
 			methods.List = append(methods.List, &ast.Field{Type: embeddedExpr})
 		}
+		if !hasComparable && typ.IsComparable() {
+			methods.List = append(methods.List, &ast.Field{Type: ast.NewIdent("comparable")})
+		}
```

Comment:
why would this happen and why is this necessary? I can't really understand the situation and the fix. at least, docs would be good.